internal/testgen/gen_negative_positive.go

```diff
@@ -0,0 +1,232 @@
+			{Fn: "LessOrEqual", Argsf: "0, a"},
+			{Fn: "LessOrEqual", Argsf: "a, 0"},
+			{Fn: "LessOrEqual", Argsf: "1, a"},
+			{Fn: "LessOrEqual", Argsf: "a, 1"},
+
+			{Fn: "True", Argsf: "a > -1"},
+			{Fn: "True", Argsf: "a < -1"},
+			{Fn: "True", Argsf: "a >= -1"},
+			{Fn: "True", Argsf: "a <= -1"},
+			{Fn: "True", Argsf: "a == -1"},
+			{Fn: "True", Argsf: "a != -1"},
+			{Fn: "True", Argsf: "-1 > a"},
+			{Fn: "True", Argsf: "-1 < a"},
+			{Fn: "True", Argsf: "-1 >= a"},
+			{Fn: "True", Argsf: "-1 <= a"},
+			{Fn: "True", Argsf: "-1 == a"},
+			{Fn: "True", Argsf: "-1 != a"},
+
+			{Fn: "True", Argsf: "a >= 0"},
+			{Fn: "True", Argsf: "a <= 0"},
+			{Fn: "True", Argsf: "a == 0"},
+			{Fn: "True", Argsf: "a != 0"},
+			{Fn: "True", Argsf: "0 >= a"},
+			{Fn: "True", Argsf: "0 <= a"},
+			{Fn: "True", Argsf: "0 == a"},
+			{Fn: "True", Argsf: "0 != a"},
+
+			{Fn: "True", Argsf: "a > 1"},
+			{Fn: "True", Argsf: "a < 1"},
+			{Fn: "True", Argsf: "a >= 1"},
+			{Fn: "True", Argsf: "a <= 1"},
+			{Fn: "True", Argsf: "a == 1"},
+			{Fn: "True", Argsf: "a != 1"},
+			{Fn: "True", Argsf: "1 > a"},
+			{Fn: "True", Argsf: "1 < a"},
+			{Fn: "True", Argsf: "1 >= a"},
+			{Fn: "True", Argsf: "1 <= a"},
+			{Fn: "True", Argsf: "1 == a"},
+			{Fn: "True", Argsf: "1 != a"},
+
+			{Fn: "False", Argsf: "a > -1"},
+			{Fn: "False", Argsf: "a < -1"},
+			{Fn: "False", Argsf: "a >= -1"},
+			{Fn: "False", Argsf: "a <= -1"},
+			{Fn: "False", Argsf: "a == -1"},
+			{Fn: "False", Argsf: "a != -1"},
+			{Fn: "False", Argsf: "-1 > a"},
+			{Fn: "False", Argsf: "-1 < a"},
+			{Fn: "False", Argsf: "-1 >= a"},
+			{Fn: "False", Argsf: "-1 <= a"},
+			{Fn: "False", Argsf: "-1 == a"},
+			{Fn: "False", Argsf: "-1 != a"},
+
+			{Fn: "False", Argsf: "a > 0"},
+			{Fn: "False", Argsf: "a < 0"},
+			{Fn: "False", Argsf: "a == 0"},
+			{Fn: "False", Argsf: "a != 0"},
+			{Fn: "False", Argsf: "0 > a"},
+			{Fn: "False", Argsf: "0 < a"},
+			{Fn: "False", Argsf: "0 == a"},
+			{Fn: "False", Argsf: "0 != a"},
+
+			{Fn: "False", Argsf: "a > 1"},
+			{Fn: "False", Argsf: "a < 1"},
+			{Fn: "False", Argsf: "a >= 1"},
+			{Fn: "False", Argsf: "a <= 1"},
+			{Fn: "False", Argsf: "a == 1"},
+			{Fn: "False", Argsf: "a != 1"},
+			{Fn: "False", Argsf: "1 > a"},
+			{Fn: "False", Argsf: "1 < a"},
+			{Fn: "False", Argsf: "1 >= a"},
+			{Fn: "False", Argsf: "1 <= a"},
+			{Fn: "False", Argsf: "1 == a"},
+			{Fn: "False", Argsf: "1 != a"},
+
+			// These one will be reported by useless-assert.
+			{Fn: "Equal", Argsf: "0, 0"},
+			{Fn: "Equal", Argsf: "a, a"},
+			{Fn: "NotEqual", Argsf: "0, 0"},
+			{Fn: "NotEqual", Argsf: "a, a"},
+			{Fn: "Greater", Argsf: "0, 0"},
+			{Fn: "Greater", Argsf: "a, a"},
+			{Fn: "GreaterOrEqual", Argsf: "0, 0"},
+			{Fn: "GreaterOrEqual", Argsf: "a, a"},
+			{Fn: "Less", Argsf: "0, 0"},
+			{Fn: "Less", Argsf: "a, a"},
+			{Fn: "LessOrEqual", Argsf: "0, 0"},
+			{Fn: "LessOrEqual", Argsf: "a, a"},
+			{Fn: "True", Argsf: "a > a"},
```

Comment:
https://github.com/Antonboom/testifylint/issues/80

Reply:
I saw you created it, thanks